sensors: add als-pt19 sensor support

dipakgmx opened the pull request:
Added [ALS-PT19 ](https://en.everlight.com/wp-content/plugins/ItemRelationship/product_files/pdf/ALS-PT19-315C-L177-TR8_V8.pdf)ambient sensor support.

The following are done in the PR:
Added new binding for the sensor
Added new sensor driver
Added basic sample to measure ambient light readings using the sensor present on the frdm_mcxw71 board.

Signed-off-by: Dipak Shetty <shetty.dipak@gmx.com>

dipakgmx committed:
dts: bindings: sensor: add everlight ambient light sensor binding

Added new binding for Everlight ALS-PT19 Ambient Light Sensor.

Signed-off-by: Dipak Shetty <shetty.dipak@gmx.com>

dipakgmx committed:
drivers: sensor: add everlight light sensor

Adds driver for Everlight ALS-PT19 Ambient Light Sensor.

Signed-off-by: Dipak Shetty <shetty.dipak@gmx.com>

yasinustunerg commented on samples/sensor/als_pt19/README.rst:
```diff
@@ -0,0 +1,42 @@
+Building and Running
+********************
+
+Currently, the sample is supported on the following boards:
+
+:zephyr:board:`frdm_mcxw71`
+
+In this example below the :zephyr:board:`frdm_mcxw71` board is used.
```
You may add sample.yaml for this sample.

See [platform_allow](https://docs.zephyrproject.org/latest/develop/test/twister.html#:~:text=platform_allow%3A%20%3Clist%20of%20platforms)

yasinustunerg commented:
> drivers: sensor: add everlight ambient light sensor binding
Added new binding for Everlight ALS-PT19 Ambient Light Sensor.

Could you change the commit message to:

> 'dts: bindings: sensor: ...'

because the only changes are in the binding file?

yperess commented on drivers/sensor/everlight/als_pt19/CMakeLists.txt:
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+zephyr_library_sources_ifdef(CONFIG_EVERLIGHT_ALS_PT19		als_pt19.c)
```
You only add this directory if  `CONFIG_EVERLIGHT_ALS_PT19` in `drivers/sensor/everlight/CMakeLists.txt` so this check is redundant. As it's written it looks like you could have an empty (no source) library.

yperess commented on drivers/sensor/everlight/als_pt19/als_pt19.c:
```diff
@@ -0,0 +1,131 @@
+	val->val2 = 0;
+
+	LOG_DBG("ADC: %d/%dmV, Current: %duA, Lux: %d.%03d", raw_val, config->adc.vref_mv,
+		current_ua, val->val1, val->val2 / 1000);
```
You're always setting val2 to 0, no need to pretty print it.

yasinustunerg commented on samples/sensor/als_pt19/src/main.c:
```diff
@@ -0,0 +1,40 @@
+		struct sensor_value lux;
+
+		read = sensor_sample_fetch(dev);
+		if (read) {
```
nit: variable name can be different. 'read' is a bit confusing. And you may declare the variables in the main scope.

jeppenodgaard reviewed (commented):
2nd commit message looks like it repeats 4 times.

Would it be possible to make grove_light a generic light sample and use that instead of creating a new specific sample?

dipakgmx commented:
> Would it be possible to make grove_light a generic light sample and use that instead of creating a new specific sample?

Actually a good idea. I had based the sample on it. I guess, I could do something like how the thermometer sample does it.

jeppenodgaard commented:
> Actually a good idea. I had based the sample on it. I guess, I could do something like how the thermometer sample does it.

Sure. I also think all the *_polling samples are generic.

MaureenHelm commented on drivers/sensor/everlight/als_pt19/als_pt19.c:
```diff
@@ -0,0 +1,131 @@
+}
+
+#define ALS_PT19_INST(inst)                                                                        \
+	static struct als_pt19_data als_pt19_data_##inst = {0};                                    \
```
```suggestion
	static struct als_pt19_data als_pt19_data_##inst;                                    \
```

MaureenHelm reviewed (changes requested):
> Would it be possible to make grove_light a generic light sample and use that instead of creating a new specific sample?

Please do this

jeppenodgaard reviewed (commented):
The new generic directory should be named `light_polling` to match the other generic polling samples.
The 3 refactor commits should be merged into one I think.

dipakgmx committed:
samples: sensor: light: add everlight light sensor

Sample extended to support the everlight light sensor on the
frdm_mcxw71 board.

Signed-off-by: Dipak Shetty <shetty.dipak@gmx.com>

dipakgmx committed:
samples: sensor: light: refactor into generic light polling sample

Moved the grove light sensor into a generic light sensor sample.
Replaced board specific node name for light sensor with an alias.
Refactored project name to reflect generic light sensor.

Signed-off-by: Dipak Shetty <shetty.dipak@gmx.com>

kartben commented on dts/bindings/sensor/everlight,als-pt19.yaml:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2025 Dipak Shetty <shetty.dipak@gmx.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Everlight Ambient Light Sensor
```
```suggestion
description: Everlight ALS-PT19 Ambient Light Sensor
```

kartben commented on samples/sensor/grove_light/sample.yaml:
this could probably be git mv'd like the other files?

dipakgmx replied:
This one is. a bit confusing. Am not sure why this is showing up as being deleted and then added. I had done a proper git mv, and its also seen in the git logs:

```
zephyr % git show --stat --oneline 1190167 
11901674a08 samples: sensor: light: refactor into generic light polling sample
 samples/sensor/{grove_light => light_polling}/CMakeLists.txt                  |  2 +-
 samples/sensor/{grove_light => light_polling}/README.rst                      | 19 ++++++++-----------
 samples/sensor/{grove_light => light_polling}/boards/nrf52dk_nrf52832.overlay |  7 ++++++-
 samples/sensor/{grove_light => light_polling}/prj.conf                        |  0
 samples/sensor/{grove_light => light_polling}/sample.yaml                     | 11 ++++++-----
 samples/sensor/{grove_light => light_polling}/src/main.c                      |  4 ++--
 6 files changed, 23 insertions(+), 20 deletions(-)
```

kartben reviewed (changes requested):
some comments on the sample part - thanks!
